src/extension/wps/wps-core/src/main/java/org/geoserver/wps/xml/WPSInternalXSDParserDelegate.java

```diff
@@ -0,0 +1,30 @@
+package org.geoserver.wps.xml;
+
+import java.lang.reflect.Field;
+
+import org.geotools.xml.Configuration;
+import org.geotools.xml.XSDParserDelegate;
+import org.geotools.xml.impl.ParserHandler;
+import org.springframework.util.ReflectionUtils;
+import org.xml.sax.helpers.NamespaceSupport;
+
+/**
+ * Extends XSDParserDelegate to make package level ParserHandler accesible via protected getter getHandler()
+ * using ReflectionUtils
+ * Adds a NamespaceSupport to default handler
+ * @author GeoSolutions
+ */
+public class WPSInternalXSDParserDelegate extends XSDParserDelegate {
+
+    public WPSInternalXSDParserDelegate(Configuration configuration, NamespaceSupport nsSupport) {
+        super(configuration);
+        getHandler().getNamespaceSupport().add(nsSupport);
+    }
+    
+    protected ParserHandler getHandler(){
+        Field field = ReflectionUtils.findField(XSDParserDelegate.class, "handler");
+        ReflectionUtils.makeAccessible(field);
+        return (ParserHandler)ReflectionUtils.getField(field, this);
```

Comment:
No need to use reflection here, as I suggested the variable should be made ``protected``. So I would:

1. Create an [issue in GeoTools](https://osgeo-org.atlassian.net/projects/GEOT/issues/GEOT-5865?filter=allopenissues) explaining why we need to access the ``handler`` variable, basically this will allow extensions to customize the handler.

2. Make a PR in GeoTools with the change, no tests required for that PR.